### PR TITLE
Add Dockerfile 

### PR DESCRIPTION
**Pull request by sbose78**
- Adding this Dockerfile until we have automated mirroring of builds happening from openshift CI
- I plan to integrate this with quay.io/redhat-developer
- Added `make build`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
+FROM centos:7 as build-tools
```
Comment (sbose78) [inline Dockerfile]: Haven't figured out how to use UBI here.

File: pkg/controller/servicebindingrequest/servicebindingrequest_controller.go [hunk 2]
```diff
@@ -157,7 +157,7 @@ outerLoop:
 					evs := &corev1.EnvVarSource{
 						SecretKeyRef: sks,
 					}
-					evn := strings.ToUpper(strings.ReplaceAll(instance.Name, "-", "_")) + "_" + strings.ToUpper(strings.ReplaceAll(key, "-", "_"))
+					evn := strings.ToUpper(strings.Replace(instance.Name, "-", "_", -1)) + "_" + strings.ToUpper(strings.Replace(key, "-", "_", -1))
```
Comment (baijum) [inline pkg/controller/servicebindingrequest/servicebindingrequest_controller.go]: This is an unrelated change. Please use Go 1.12

File: pkg/controller/servicebindingrequest/servicebindingrequest_controller.go [hunk 3]
```diff
@@ -173,7 +173,7 @@ outerLoop:
 					evs := &corev1.EnvVarSource{
 						ConfigMapKeyRef: cmks,
 					}
-					evn := strings.ToUpper(strings.ReplaceAll(instance.Name, "-", "_")) + "_" + strings.ToUpper(strings.ReplaceAll(key, "-", "_"))
+					evn := strings.ToUpper(strings.Replace(instance.Name, "-", "_", -1)) + "_" + strings.ToUpper(strings.Replace(key, "-", "_", -1))
```
Comment (baijum) [inline pkg/controller/servicebindingrequest/servicebindingrequest_controller.go]: This is an unrelated change. Please use Go 1.12

File: Makefile [hunk 4]
```diff
@@ -141,3 +141,9 @@ test-unit:
 ./vendor: go.mod go.sum
 	$(Q)GOCACHE=$(shell pwd)/out/gocache GO111MODULE=on go mod vendor ${V_FLAG}
 
+
+.PHONY: build
+build: ./out/operator
+
+./out/operator:
+	$(Q)CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build  ${V_FLAG} -o ./out/operator  cmd/manager/main.go
```
Comment (baijum) [inline Makefile]: `operator-sdk build` produce docker image with UBI as the base image.
Earlier we used `go build` as there was no way pass build flags. But that is fixed in SDK: https://github.com/operator-framework/operator-sdk/issues/1435

File: pkg/controller/servicebindingrequest/servicebindingrequest_controller.go (hunk 2 above)
Comment (sbose78) [reply pkg/controller/servicebindingrequest/servicebindingrequest_controller.go]: Ah ok! That explains it. Thanks.

**Comment by sbose78** [comment]
Ok, where could I find the operator image built off master?

**Comment by baijum** [comment]
> Ok, where could I find the operator image built off master?

As soon as you run `operator-sdk build <image>` command, it produces the image. The image will be in the same system where this command was run.

But if the intent of this Makefile target is to use inside OpenShift CI like how we did it in DevConsole operator, then this will be required.
https://github.com/openshift/release/blob/872b1e09f6e5c3cca821b8c8bac90b54f9355573/ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml#L23

**Comment by baijum** [comment]
@sbose78 You can use the `openshift-ci/Dockerfile.tools` for the Dockerfile here. It has code Go 1.12 and other required toold.

**Comment by sbose78** [comment]
Eventually the intent is to build it on ci and mirror images to Quay. 

But till we have that, I wish to setup master builds on Quay so that it could be used in the OperatorSource without having to locally build it and push it.

**Comment by sbose78** [comment]
( I will address the go1.12 comments )

**Comment by otaviof** [comment]
If we really need a Dockerfile to be able to run this under OpenShift CI, we should find a automated way to extract the data from a `operator-sdk build` command output. By using automation we can mitigate the risk of deviating from what `operator-sdk` provides for the images we will be consuming in upstream.

**Comment by baijum** [comment]
@otaviof I have sent a PR to introduce building without OCI image: https://github.com/operator-framework/operator-sdk/pull/1600
